crypto/pkcs12/p12_key.c

```diff
@@ -68,16 +70,32 @@ int PKCS12_key_gen_uni(unsigned char *pass, int passlen, unsigned char *salt,
                        int saltlen, int id, int iter, int n,
                        unsigned char *out, const EVP_MD *md_type)
 {
-    unsigned char *B = NULL, *D = NULL, *I = NULL, *p = NULL, *Ai = NULL;
-    int Slen, Plen, Ilen;
-    int i, j, u, v;
-    int ret = 0;
-    EVP_MD_CTX *ctx = NULL;
-    unsigned char *tmpout = out;
-    int tmpn = n;
+    int res = 0;
+    EVP_KDF *kdf;
+    EVP_KDF_CTX *ctx;
+    OSSL_PARAM params[6], *p = params;
 
-    ctx = EVP_MD_CTX_new();
+    if (n <= 0)
+        return 0;
+
+    kdf = EVP_KDF_fetch(NULL, "PKCS12KDF", NULL);
```

Comment:
Is this a case where the libctx will be injected if required..

Reply:
The public API doesn't have access to the libctx and didn't use it to fetch the digest.
Adding new public APIs is possible but is it worthwhile?

Reply:
I dont know the answer to that. If we are moving the KDF into the proper place then maybe it should have the with_libctx variant (but then it will need it for the other cases also).

Reply:
Hmm, if the md_type passed into the call was a fetched md we could infer the libctx from the OSSL_PROVIDER in it.

Reply:
Good point, it does assume that the provider of the digest is the same as for this.  I'm not sure that's a safe assumption.

Reply:
Ignore my last comment, the libctx is good.  We'll still have lost the property query.

Reply:
libctx change is pushed.

Reply:
and passes CI checks.